cli/project/files.go

```diff
@@ -152,47 +167,114 @@ func getPath(conf map[string]interface{}, opts PathOpts) (string, error) {
 	return path, nil
 }
 
-// SetLocalRunningPaths fills {Run,Data,Log,Conf}Dir
-// Values are collected from specified flags and .cartridge.yml
+func setRunningConfPath(ctx *context.Ctx) error {
+	// used in unit tests
+	if ctx.Running.ConfPath != "" {
+		return nil
+	}
+
+	if ctx.Running.Global {
+		homeDir, err := common.GetHomeDir()
+		if err != nil {
+			log.Warnf("Failed to get home dir: %s, using the default %s", err, defaultHomeDir)
+			homeDir = defaultHomeDir
+		}
+
+		ctx.Running.ConfPath = filepath.Join(homeDir, runningConfFilename)
+	} else {
+		curDir, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("Failed to get current directory: %s", err)
+		}
+
+		ctx.Running.ConfPath = filepath.Join(curDir, runningConfFilename)
+	}
+
+	return nil
+}
+
+func parseRunningConf(ctx *context.Ctx) (RunningConf, error) {
+	if err := setRunningConfPath(ctx); err != nil {
+		return nil, fmt.Errorf("Failed to set running conf path: %s", err)
+	}
+
+	conf := make(RunningConf)
+
+	if _, err := os.Stat(ctx.Running.ConfPath); err == nil {
+		if conf, err = common.ParseYmlFile(ctx.Running.ConfPath); err != nil {
+			return nil, fmt.Errorf("Failed to read configuration from file: %s", err)
+		}
+	} else if !os.IsNotExist(err) {
+		return nil, fmt.Errorf("Failed to use conf file: %s", err)
+	}
+
+	return conf, nil
+}
+
+// SetRunningPaths fills {Run,Data,Log,Conf,Apps,App}Dir
+// Values are collected from specified flags and (if useConf is true)
+// * ./.cartridge.yml for local running
+// * ~/.cartridge.yml for global running
 //
 // The priority of sources is:
 // * user-specified flags
-// * value from .cartridge.yml
+// * value from .cartridge.yml (if useConf is true)
 // * default values (defined here in const section)
-func SetLocalRunningPaths(ctx *context.Ctx) error {
+func SetRunningPaths(ctx *context.Ctx, useConf bool) error {
```

Comment:
Seems it could be splitted to `SetLocalRunningPaths ` and `SetGlobalRunningPaths `. If here is quite big

Reply:
Common part could be moved into separate function

Reply:
`useConf ` variable isn't about local/global, it's about running/pack (in pack rpm and deb the same global paths are used).

The main difference of `SetLocalRunningPaths` and `SetGlobalRunningPaths` is the default values and `App(s)Dir` variable. And I sure that in the one functions it's quite simple to understand the difference and support this code.